source/elens.f90

```diff
@@ -107,217 +183,527 @@ subroutine elens_parseInputLine(inLine, iLine, iErr)
-    tmpi1 = 9
-    tmpi2 = 10
-    tmpi3 = 11
-    elens_lThetaR2(ielens(iElem)) = .true.
-  end if
-
-  if(elens_lThetaR2(ielens(iElem))) then
-    call chr_cast(lnSplit(tmpi1),elens_len(ielens(iElem)),iErr)
-    call chr_cast(lnSplit(tmpi2),elens_I(ielens(iElem)),  iErr)
-    call chr_cast(lnSplit(tmpi3),elens_Ek(ielens(iElem)), iErr)
-  end if
-
-  ! sanity checks
-  if(elens_r2(ielens(iElem)) < elens_r1(ielens(iElem))) then
-    write(lout,"(a)") "ELENS> WARNING ELEN R2<R1. Inverting."
-    tmpflt=elens_r2(ielens(iElem))
-    elens_r2(ielens(iElem)) = elens_r1(ielens(iElem))
-    elens_r1(ielens(iElem)) = tmpflt
-  else if(elens_r2(ielens(iElem)) == elens_r1(ielens(iElem))) then
-    write(lerr,"(a)") "ELENS> ERROR ELEN R2=R1. Elens does not exist."
-    iErr = .true.
-    return
-  end if
-  if(elens_r2(ielens(iElem)) <= zero) then
-    write(lerr,"(a)") "ELENS> ERROR R2<=0!"
-    iErr = .true.
-    return
-  end if
-  if(elens_r1(ielens(iElem)) < zero) then
-    write(lerr,"(a)") "ELENS> ERROR R1<0!"
-    iErr = .true.
-    return
-  end if
-  if(elens_lThetaR2(ielens(iElem))) then
-    if(elens_len(ielens(iElem)) <= zero) then
-      write(lerr,"(a)") "ELENS> ERROR L<0!"
+    
+    if(kz(iElem) /= elens_kz) then
+      write(lerr,"(3(a,i0))") "ELENS> ERROR Element type is kz(",iElem,") = ",kz(iElem)," != ",elens_kz
       iErr = .true.
       return
     end if
-    if(elens_I(ielens(iElem)) == zero) then
-      write(lerr,"(a)") "ELENS> ERROR I=0!"
+    if(el(iElem) /= zero .or. ek(iElem) /= zero .or. ed(iElem) /= zero) then
+      write(lerr,"(a)")       "ELENS> ERROR Length el(iElem) (elens is treated as thin element), "//&
+        "and first and second field have to be zero:"
+      write(lerr,"(2(a,i0),a)") "ELENS>       el(",iElem,") = ",el(iElem)," != 0"
+      write(lerr,"(2(a,i0),a)") "ELENS>       ed(",iElem,") = ",ed(iElem)," != 0"
+      write(lerr,"(2(a,i0),a)") "ELENS>       ek(",iElem,") = ",ek(iElem)," != 0"
       iErr = .true.
       return
     end if
-    if(elens_Ek(ielens(iElem)) <= zero) then
-      write(lerr,"(a)") "ELENS> ERROR Ek<0! (e-beam)"
+    
+    if(ielens(iElem) /= 0) then
+      write(lerr,"(a)") "ELENS> ERROR The element '"//trim(bez(iElem))//"' was defined twice."
       iErr = .true.
       return
     end if
-  end if
-  if( elens_type(ielens(iElem)) == 2 ) then
-    if ( elens_sig(ielens(iElem)).le.zero ) then
-       write(lerr,"(a)") "ELENS> ERROR sigma of electron beam <=0 in Elens '"//trim(bez(iElem))//"'."
-       iErr = .true.
-       return
+    nelens = nelens+1
+    call elens_expand_arrays_lenses(nelens)
+    ielens(iElem) = nelens
+    
+    ! Parse the element
+    select case (lnSplit(2))
+    case("UNIFORM")
+      elens_type(ielens(iElem)) = 1
+    case("GAUSSIAN")
+      elens_type(ielens(iElem)) = 2
+      if(nSplit < 8) then
+        write(lerr,"(a,i0)") "ELENS> ERROR Expected at least 8 input parameters for GAUSSIAN, got ",nSplit
+        iErr = .true.
```

Comment:
If the keyword is the second element, then there are 6 parameters for the keyword.